### PR TITLE
Update to draft 19

### PR DESCRIPTION
**Pull request by daxpedda**
This updates VOPRF to be in sync with draft 19.

There are no changes compared to draft 11, which is what we were on until now, except:
- The Ciphersuite ID is now a string and not two bytes anymore.
- The static draft identifier in the context string is now `OPRFV1-`.

The change to the Ciphersuite ID requires us to update to `elliptic-curve` v0.13.0-pre.5, which I've done here. This also requires an upgrade to Rust 1.61.

This small change posed some difficulties because of the non-fixed-size nature of this ID. To avoid allocation I had to change the API in `elliptic-curve` (https://github.com/RustCrypto/traits/pull/1238) and introduce a new internal helper type: `Dst`.

This also changed some of hash traits we used, namely instead of `Digest` we no require `Default + FixedOutput + HashMarker`, which should pose no downstream issues.

This build on top of:
- `elliptic-curve`: https://github.com/RustCrypto/traits/pull/1238
- `p256` (for testing only): https://github.com/RustCrypto/elliptic-curves/pull/732

~~We should probably wait for the release of `elliptic-curve` v0.13.0-pre.4 before merging this.~~
Was released!

Builds on top of https://github.com/facebook/voprf/pull/102.

### CODE REVIEW

**Comment by raphaelrobert** [comment]
There also was a change in the test in the test vectors between draft-16 and draft-17: 

https://www.google.com/url?sa=j&url=https%3A%2F%2Fauthor-tools.ietf.org%2Fiddiff%3Furl1%3Ddraft-irtf-cfrg-voprf-16%26url2%3Ddraft-irtf-cfrg-voprf-17%26difftype%3D--html&uct=1528330230&usg=tqYv9g34xelr7XxjhiOAi96Docs.&source=meet

**Comment by daxpedda** [comment]
The tests didn't change, just the output, because of the change to the context string.
For a moment I was thinking: how the hell did CI pass if I forgot to update the test vectors :smile:.

**Comment by daxpedda** [comment]
This is now ready to be merged, but if possible I would like to get a new pre-release for `voprf` in beforehand.

**Comment by daxpedda** [comment]
Now builds on top of https://github.com/facebook/voprf/pull/102.

**Comment by daxpedda** [comment]
This is ready to be merged now!

File: src/common.rs [hunk 1]
```diff
@@ -497,16 +497,64 @@ where
         .finalize())
 }
 
-/// Generates the contextString parameter as defined in
-/// <https://datatracker.ietf.org/doc/draft-irtf-cfrg-voprf/>
-pub(crate) fn create_context_string<CS: CipherSuite>(mode: Mode) -> GenericArray<u8, U11>
+pub(crate) struct Dst<L: ArrayLength<u8>> {
+    dst_1: GenericArray<u8, L>,
+    dst_2: &'static str,
+}
```
Comment (kevinlewi) [inline src/common.rs:503]: Nice! Thanks for this.

**Review by kevinlewi — APPROVED** [review]
Thanks once again for your contributions!

**Review by kevinlewi — APPROVED** [review]

**Pull request merged**